packages/appsync-modelgen-plugin/src/utils/fieldUtils.ts

```diff
@@ -13,3 +14,22 @@ export function removeFieldFromModel(model: CodeGenModel, fieldName: string): vo
 
 export const getDirective = (fieldOrModel: CodeGenField | CodeGenModel) => (directiveName: string): CodeGenDirective | undefined =>
   fieldOrModel.directives.find(d => d.name === directiveName);
+
+export const getOtherSideBelongsToFieldName = (type: string, otherSideModel: CodeGenModel): string | undefined =>
+  otherSideModel.fields.filter(f => f.type === type).find(f => f.directives.find(d => d.name === TransformerV2DiretiveName.BELONGS_TO))?.name;
```

Comment:
Could this simplified to `getOtherSideBelongsToField(type, otherSideModel)?.name`